status-im/status-react
# 💾 Save address in send flow

fixes #19546 
fixes #16994 
fixes #16995 

### Summary

This PR almost completes the saved address from send flow epic. There is one missing piece where we need to handle saving ENS names instead of HEX addresses. 

There are two major aspects that need to be tested:
1. The "Save Address" button should show on the transaction progress screen, only if the address is not saved already
2. The flow of actually saving the address
2.1 The sub flow of saving the address on different networks.


### Review notes
This PR is heavy from QA perspective. So I have hidden the entry point to the save address feature.
The idea is that we get this merged. Then extract the `network-preferences` component and make it pure. Right now it can be pure or impure. 

And when the network preferences is merged, we enable this feature in a separate PR. 

--
The other option is that the QA team prioritises this and gets it tested. So we merged it, and then work on purification of the `network-preferences` component. I'll talk to the team and update this comment accordingly. 

### Testing notes
- In order to test the flow, you'll have to transfer a token to some address.
- On the transaction progress screen, you'll see a button to save the address
- If the address is already saved, you should see no button

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- Send flow
- Wallet

##### Functional
- wallet / transactions
- dapps / app browsing


### Steps to test

First Flow (Recipient address is not saved)
- Open the app
- Goto an account with some funds
- Transfer the funds to an address
- On the transaction progress screen, you should see a button to save the address
- Click the button and finish the save address flow

Second flow (Recipient address is saved already)
- When you make the transfer, the ''Save Address'' button should not show up

status: ready
